File: BaekJoon Online Judge/C++/Step08-Math_1 Mathematical Thinking/2839.cpp
//
//  2839.cpp
//  Step08-Math_1 Mathematical Thinking
//
//  Created by Yebin Kim on 2019/10/23.
//  Copyright © 2019 김예빈. All rights reserved.
//

#include <iostream>

using namespace std;

// Time Complexity O(n*log(n))
int main() {
    cin.tie(NULL);
    ios::sync_with_stdio(false);
    
    int n, five, three = 0;
    int answer;
    
    cin >> n;
    
    five = n / 5;
    
    while(five >= 0) {
        three = n - (five * 5);
        
        if(three % 3 == 0) {
            three /= 3;
            break;
        } else
            five--;
    }
    
    if(five < 0)
        answer = -1;
    else
        answer = five + three;
    
    cout << answer;
    
    return 0;
}

File: BaekJoon Online Judge/Swift/Step08-Math_1 Mathematical Thinking/2839.swift
//
//  2839.swift
//  Step08-Math_1 Mathematical Thinking
//
//  Created by Yebin Kim on 2019/11/13.
//  Copyright © 2019 김예빈. All rights reserved.
//

import Foundation

if let N = Int(readLine()!) {
    var three = 0
    var five = N / 5
    
    while(five >= 0) {
        three = N - (five * 5)
        
        if three % 3 == 0 {
            three /= 3
            break
        } else {
            five -= 1
        }
    }
    
    if five < 0 {
        print(-1)
    } else {
        print(five + three)
    }
}
